exp/api/v1beta1/machinepool_types.go

```diff
@@ -59,6 +59,13 @@ type MachinePoolSpec struct {
 	// FailureDomains is the list of failure domains this MachinePool should be attached to.
 	// +optional
 	FailureDomains []string `json:"failureDomains,omitempty"`
+
+	// Whether the Replicas value is externally managed. This is useful when
+	// the infrastructure is a managed machine pool, or if the cluster-autoscaler is scaling
+	// the underlying cloud infrastructure directly outside of cluster-api. Set this to true
+	// if you wish the replica count to be managed outside of cluster-api. Defaults to false.
+	// +optional
+	ExternallyManagedReplicaCount *bool `json:"externallyManagedReplicaCount,omitempty"`
```

Comment:
Is this field used anywhere?

Reply:
It will be used to signal to the infrastructure provider that it is responsible for managing the `spec.replicas` and relevant status fields on the MachinePool. So there is no implementation code in the MachinePool itself yet. That may change a little as the MachinePool Machines implementation begins though.